api/v1alpha1/kubernetes_helpers.go

```diff
@@ -112,3 +117,40 @@ func (hpa *KubernetesHorizontalPodAutoscalerSpec) setDefault() {
 		hpa.Metrics = DefaultEnvoyProxyHpaMetrics()
 	}
 }
+
+// ApplyMergePatch applies a merge patch to a deployment based on the merge type
+func (deployment *KubernetesDeploymentSpec) ApplyMergePatch(old *appv1.Deployment) (*appv1.Deployment, error) {
```

Comment:
IMO, this more like a util function.

Reply:
Yes, since ratelimit and proxy are both using it, keeping it in kubernetes_helpers is a common util too.

Reply:
I would also prefer if we moved this out of the `api` directory into https://github.com/envoyproxy/gateway/tree/main/internal/kubernetes